lich.rbw

```diff
@@ -8736,20 +8758,40 @@ module Effects
       @dialog = dialog
     end
 
-    def to_h
-      XMLData.dialogs.fetch(@dialog, {})
+    def to_h(complex: false)
+      if complex
+        XMLData.dialogs.fetch(@dialog, { :names => {}, :data => {} })
+      else
+        XMLData.active_spells
+    end
     end
 
     def each()
       to_h.each {|k,v| yield(k,v)}
     end
 
-    def active?(effect)
-      expiry = to_h.fetch(effect, 0)
+    def active?(id_or_name)
+      begin
+        expiry = to_h[:data].fetch(id_or_name, 0) if id_or_name.instance_of?(Integer)
+        expiry = to_h[:data].fetch(to_h[:names].transform_keys(&:downcase).fetch(id_or_name.downcase), 0) if id_or_name.instance_of?(String)
+        expiry = to_h[:data].fetch(to_h[:names].transform_keys(&:downcase).fetch(id_or_name.to_s.gsub(/[\_]/, ' ').split.map(&:downcase).join(' ')), 0) if id_or_name.instance_of?(Symbol)
       expiry.to_i > Time.now.to_i
+      rescue
+        false
     end
   end
 
+  def temp_putup(effect:)
```

Comment:
This is an unnecessary addition if the breaking changes are reverted

Reply:
If I can successfully incorporate the suggested simplifications, I will test - however, it is unclear how even the suggested simplifications will address a lag event or delay from SIMU providing the XML data.  This holds true today in 5.0.16, and I would expect holds true with any changes in 5.0.18 as it is dependent on SIMU timing, not Lich coordination.  This is likely to stay unless we decide it should be a script responsibility and not exposed in core Lich.

Reply:
I will leave open the question around managing SIMU xml data delays (game lag), but maintain this may become something we want to include.  I do not see the direct correlation between this and breaking changes, but will continue to explore.